client/state/data-layer/wpcom/sites/plan-transfer/index.js

```diff
@@ -0,0 +1,70 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { SITE_PLAN_OWNERSHIP_TRANSFER } from 'state/action-types';
+
+const noticeOptions = siteId => ( {
+	duration: 8000,
+	id: `sites-plan-transfer-notice-${ siteId }`,
+} );
+
+/**
+ * Dispatches a request to transfer a site's plan to another user.
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const requestPlanOwnershipTransfer = action =>
+	http(
+		{
+			apiVersion: '1',
+			method: 'POST',
+			path: '/sites/' + action.siteId + '/plan-transfer',
+			query: {
+				new_user_id: action.newOwnerUserId,
+			},
+		},
+		action
+	);
+
+/**
+ * Dispatches a success notice when the request succeeded.
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Success notice action
+ */
+export const handleTransferSuccess = ( { siteId } ) =>
+	successNotice(
+		translate( 'Plan purchaser has been changed successfully.' ),
+		noticeOptions( siteId )
+	);
+
+/**
+ * Dispatches an error notice when the request failed.
+ *
+ * @param   {Object} action Redux action
+ * @param   {Object} error  Error object
+ * @returns {Object} Error notice action
+ */
+export const handleTransferError = ( { siteId }, { message } ) =>
+	errorNotice( message, noticeOptions( siteId ) );
```

Comment:
can we work hard on sending some specific error codes from the server and displaying relevant context to the customer here? _why_ did it fail?

not that we can also detect the difference between a failed network request and a successful one that failed on the API call.

check the credentials updating to see one way that we are providing that context _and_ an in-context link to Happychat

Reply:
I'm not a huge fan of error codes in customer-facing messages. We've intentionally kept the messages customer-facing and more user-friendly, while they still explain what exactly happened. We can always iterate with that, but for a v1 I'm pretty happy with the information we provide when an error occurs at this point.

Reply:
My point wasn't to expose error codes. If you check the credentials form you can see that we use error codes to send helpful contextual messages to customers to empower them to know what happened and what action steps they can take as a response. Here at best we're sending a string from PHP which probably won't be written in the language Calypso is using.

Iteration is good. Let's make sure Happiness is tracking closely with this stuff. I'm pretty sure they prefer some context so that people don't end up coming their way and saying, "Your website didn't work." 😉

Reply:
> Let's make sure Happiness is tracking closely with this stuff. 

Yes! That's why we're doing this after all, right 😉 They'll definitely know about it when we release it to public.